### DISCUSSION TITLE
How to install nvidia-l4t-jetson-io_35.4.1 in yocto image?

### DISCUSSION THREAD

**Discussion by fengyingchu0827** [OE4T/meta-tegra#1422] [Q&A] [answered]
Hi,
We would like to use config-by-hardware.py of nvidia-l4t-jetson-io_35.4.1.
We cannot find any related bb file in the folder of /repos/meta-tegra/recipes-bsp/tegra-binaries.

The branch we use is kirkstone. May we ask how to have nvidia-l4t-jetson-io_35.4.1 at our yocto image?

Thanks a lot.

**Comment by madisongh** [comment]
We don't provide a recipe for that package because I never found a good use case for it... With NVIDIA's Jetson Linux, they provide everything in pre-built form and has to allow for some customization at runtime. With Yocto, you control the build of your custom OS and can configure the device tree at build time for the hardware you are using.

What were you planning to use jetson-io for?

**Reply by fengyingchu0827** [reply]
Hi @madisongh ,
we want to create a Device Tree overlay for a custom hardware module, referring to the website: https://developer.ridgerun.com/wiki/index.php/NVIDIA_Jetson_-_Device_Tree_Overlay 

We would like to apply the device tree overlay created, for example by using: sudo /opt/nvidia/jetson-io/config-by-hardware.py -n "My Jetson Overlay example"

Thank you very much.

**Reply by dwalkes** [reply]
@fengyingchu0827 see notes at https://github.com/OE4T/meta-tegra/wiki/Using-device-tree-overlays about using `KERNEL_DEVICETREE_APPLY_OVERLAYS` assuming you can apply at build time instead of run time

**Reply by fengyingchu0827** [reply]
Hi @dwalkes ,
unfortunately, we need to apply the device tree overlay at run time, instead of build time. 
The feature of config-by-hardware.py of the jetson-io package could provide us a convenient way to have the new property defined in the device tree using the Device Tree overlay.

**Reply by madisongh** [reply, marked as answer]
At build time you would:
1. Create a second `.dtb` file with your device tree applied to it, as described on that RidgeRun page you mentioned
2. Modify the creation of the `extlinux.conf` file to create a second entry with the second device tree

That would give users the runtime choice of booting with either the default DTB or the one with your overlay applied, but the configuration choices would be generated at build time.

**Reply by fengyingchu0827** [reply]
@madisongh 
Thank you. We use fdtoverlay command to apply dtbo into the dtb file. We appreciate your support.
